librosa/util/_nnls.py

```diff
@@ -25,13 +25,13 @@ def _nnls_obj(x, shape, A, B):
     x = x.reshape(shape)
 
     # Compute the difference matrix
-    diff = np.dot(A, x) - B
+    diff = np.einsum('mf,...ft->...mt',A,x, optimize=True) - B
 
     # Compute the objective value
-    value = 0.5 * np.sum(diff ** 2)
+    value = 0.5 * np.sum(diff ** 2, axis=(-1,-2))
```

Comment:
do we need target axes here?  value should be scalar.

Reply:
I thought that keeping the sum for each channel independently would help to make my test pass. In any case, the results are the same. I guess the optimizer does a summation on its own?